sc_cost_meter/app.py

```diff
@@ -1,42 +1,39 @@
-import json
-
-# import requests
+import logging
+import sc_cost_meter.utils as utils
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def meter_instances():
+  '''
+  Send EC2 cost info to the AWS Marketplace
+  '''
+  instances = utils.get_instances(['running'])
+  for instance in instances:
+    instance_id = instance['InstanceId']
+    tags = utils.get_tags(instance_id)
+    for tag in tags:
+      if tag['Key'] == 'marketplace:customerId':
+        customer_id = tag['Value']
+        product_code = utils.get_marketplace_product_code(tags)
+        if product_code:
+          cost_tags = {
+            "Key": "marketplace:customerId",
+                  "Values": [
+                     customer_id
+                  ]
+          }
+          cost = utils.get_ec2_cost(cost_tags)
```

Comment:
This looks wrong to me:  It looks like for each running EC2 we retrieve all costs for a customer (across all EC2s) and send a bill. So if a customer has 10 EC2s then we will bill for all 10 EC2 10 times.  

There should be a test in which there is >1 instance, each with a different amount to bill, and we check that we call `utils.report_usage` with the right total.

Reply:
hmm, will create some to tests to verify.

Reply:
thanks for catching.  My approach to getting the cost was completely off base.  I have refactored to get cost for all resources and i think i've made the logic for this much simpler.  Also added unit tests as well.